cmd/kubeadm/app/cmd/init.go

```diff
@@ -461,6 +468,34 @@ func (d *initData) Tokens() []string {
 	return tokens
 }
 
+// TryLoadCertFromDisk tries to get cert from cache or disk
+func (d *initData) TryLoadCertFromDisk(pkiPath, name string) (*x509.Certificate, error) {
```

Comment:
Not sure if this is the best place to store this logic. This should probably be moved to `pkiutil` and be available to everyone.

Reply:
i agree, +1 to moving outside of init.go

Reply:
but this is tricky.
the values from map (as a collection) is probably not something that the rest of the code base needs.

Reply:
Maybe, that's why I am not sure about this, but we can do a research in the future.